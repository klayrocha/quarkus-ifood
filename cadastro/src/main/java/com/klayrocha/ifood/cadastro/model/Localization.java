package com.klayrocha.ifood.cadastro.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "localization")
public class Localization {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	public Long id;

	public Double latitude;

	public Double longitude;
}
