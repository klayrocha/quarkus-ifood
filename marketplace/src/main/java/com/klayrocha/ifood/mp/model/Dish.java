package com.klayrocha.ifood.mp.model;

import java.math.BigDecimal;
import java.util.stream.StreamSupport;

import com.klayrocha.ifood.mp.dto.DishDTO;

import io.smallrye.mutiny.Multi;
import io.smallrye.mutiny.Uni;
import io.vertx.mutiny.pgclient.PgPool;
import io.vertx.mutiny.sqlclient.Row;
import io.vertx.mutiny.sqlclient.RowSet;
import io.vertx.mutiny.sqlclient.Tuple;

public class Dish {

	public Long id;

	public String name;

	public String description;

	public Restaurant restaurant;

	public BigDecimal price;

	public static Multi<DishDTO> findAll(PgPool pgPool) {
		Uni<RowSet<Row>> preparedQuery = pgPool.query("select * from dish").execute();
		return unitToMulti(preparedQuery);
	}

	public static Multi<DishDTO> findAll(PgPool client, Long idRestaurant) {
		Uni<RowSet<Row>> preparedQuery = client
				.preparedQuery("SELECT * FROM dish where dish.restaurant_id = $1 ORDER BY name ASC")
				.execute(Tuple.of(idRestaurant));
		return unitToMulti(preparedQuery);
	}

	private static Multi<DishDTO> unitToMulti(Uni<RowSet<Row>> queryResult) {
		return queryResult.onItem().transformToMulti(set -> Multi.createFrom().items(() -> {
			return StreamSupport.stream(set.spliterator(), false);
		})).onItem().transform(DishDTO::from);
	}

	public static Uni<DishDTO> findById(PgPool client, Long id) {
		return client.preparedQuery("SELECT * FROM dish WHERE id = $1").execute(Tuple.of(id)).map(RowSet::iterator)
				.map(iterator -> iterator.hasNext() ? DishDTO.from(iterator.next()) : null);
	}

}
