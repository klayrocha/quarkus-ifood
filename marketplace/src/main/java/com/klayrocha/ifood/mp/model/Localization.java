package com.klayrocha.ifood.mp.model;

public class Localization {

	public Long id;

	public Double latitude;

	public Double longitude;

	@Override
	public String toString() {
		return "Localization [id=" + id + ", latitude=" + latitude + ", longitude=" + longitude + "]";
	}

}
