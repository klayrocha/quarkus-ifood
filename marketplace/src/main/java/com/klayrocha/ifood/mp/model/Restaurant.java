package com.klayrocha.ifood.mp.model;

import io.vertx.mutiny.pgclient.PgPool;
import io.vertx.mutiny.sqlclient.Tuple;

public class Restaurant {

	public Long id;

	public String name;

	public Localization localization;

	public void persist(PgPool pgPool) {
		pgPool.preparedQuery("insert into localization (id, latitude, longitude) values ($1, $2, $3)")
				.execute(Tuple.of(localization.id, localization.latitude, localization.longitude)).await()
				.indefinitely();

		pgPool.preparedQuery("insert into restaurant (id, name, localization_id) values ($1, $2, $3)")
				.execute(Tuple.of(id, name, localization.id)).await().indefinitely();

	}

	@Override
	public String toString() {
		return "Restaurant [id=" + id + ", name=" + name + ", localization=" + localization + "]";
	}

}
