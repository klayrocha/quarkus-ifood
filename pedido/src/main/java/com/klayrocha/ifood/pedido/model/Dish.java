package com.klayrocha.ifood.pedido.model;

import org.bson.types.Decimal128;

public class Dish {

	public String name;

	public String description;

	public Decimal128 price;

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public Decimal128 getPrice() {
		return price;
	}

	public void setPrice(Decimal128 price) {
		this.price = price;
	}
	
	

}
