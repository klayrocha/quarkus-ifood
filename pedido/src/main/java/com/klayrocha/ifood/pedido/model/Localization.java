package com.klayrocha.ifood.pedido.model;

public class Localization {
	
	public Double latitude;

    public Double longitude;

	public Double getLatitude() {
		return latitude;
	}

	public void setLatitude(Double latitude) {
		this.latitude = latitude;
	}

	public Double getLongitude() {
		return longitude;
	}

	public void setLongitude(Double longitude) {
		this.longitude = longitude;
	}

    
    
}
