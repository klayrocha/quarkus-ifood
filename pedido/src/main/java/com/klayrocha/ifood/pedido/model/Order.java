package com.klayrocha.ifood.pedido.model;

import java.util.List;

import io.quarkus.mongodb.panache.MongoEntity;
import io.quarkus.mongodb.panache.PanacheMongoEntity;

@MongoEntity(collection = "orders", database = "admin")
public class Order extends PanacheMongoEntity {

    public String customer;

    public List<Dish> dishes;

    public Restaurant restaurant;

    public String deliveryman;

    public Localization localizationDeliveryman;

	public String getCustomer() {
		return customer;
	}

	public void setCustomer(String customer) {
		this.customer = customer;
	}

	public List<Dish> getDishes() {
		return dishes;
	}

	public void setDishes(List<Dish> dishes) {
		this.dishes = dishes;
	}

	public Restaurant getRestaurant() {
		return restaurant;
	}

	public void setRestaurant(Restaurant restaurant) {
		this.restaurant = restaurant;
	}

	public String getDeliveryman() {
		return deliveryman;
	}

	public void setDeliveryman(String deliveryman) {
		this.deliveryman = deliveryman;
	}

	public Localization getLocalizationDeliveryman() {
		return localizationDeliveryman;
	}

	public void setLocalizationDeliveryman(Localization localizationDeliveryman) {
		this.localizationDeliveryman = localizationDeliveryman;
	}
    
    

}