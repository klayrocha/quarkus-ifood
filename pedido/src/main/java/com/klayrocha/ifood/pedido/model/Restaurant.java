package com.klayrocha.ifood.pedido.model;

public class Restaurant {

	 public String name;

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
	 
	 
	 
}
