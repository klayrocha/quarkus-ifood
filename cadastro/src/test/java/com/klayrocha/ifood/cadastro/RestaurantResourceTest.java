package com.klayrocha.ifood.cadastro;

import javax.ws.rs.core.Response.Status;

import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.klayrocha.ifood.cadastro.model.Restaurant;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifeCycleManager.class)
public class RestaurantResourceTest {

    @Test
    @DataSet("restaurant-scene.yml")
    public void testfindRestaurants() {
        String result = given()
          .when().get("/restaurants")
          .then()
             .statusCode(200)
             .extract().asString();
        
        Approvals.verifyJson(result);
    }
    
    private RequestSpecification given() {
    	return RestAssured.given().contentType(ContentType.JSON);
    }

    
    @Test
    @DataSet("restaurant-scene.yml")
    public void testUpdateRestaurants() {
    	
    	Restaurant dto = new Restaurant();
    	dto.name = "newName";
    	
    	Long parameterValue = 123L;
    	
        given()
          .when().pathParam("id", parameterValue)
          .body(dto)
          .when().put("/restaurants/{id}")
          .then()
             .statusCode(Status.NO_CONTENT.getStatusCode())
             .extract().asString();
        
        Restaurant findById = Restaurant.findById(parameterValue);
        
        Assert.assertEquals(dto.name, findById.name);
    }

}