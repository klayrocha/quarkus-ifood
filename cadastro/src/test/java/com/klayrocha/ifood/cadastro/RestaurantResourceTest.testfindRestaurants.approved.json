[
  {
    "id": 123,
    "name": "New Restaurant",
    "owner": "ID do Keyclock"
  },
  {
    "id": 234,
    "name": "New Restaurant 2",
    "owner": "ID do Keyclock 2"
  }
]